docs/rst/tutorials/micro/custom_transports/custom_transports.rst

```diff
@@ -0,0 +1,279 @@
+.. _tutorials_micro_custom_transports:
+
+Custom Transports
+=================
+
+.. contents:: Table of Contents
+    :depth: 1
+    :local:
+    :backlinks: none
+
+
+This tutorial aims at providing step-by-step guidance for those users interested in creating micro-ROS custom transports, instead of using the ones provided by default in the micro-ROS tools set.
+
+This tutorial starts from a previously created micro-ROS environment. Check the first steps of `First micro-ROS application on an RTOS <https://docs.vulcanexus.org/en/latest/rst/tutorials/micro/getting_started/getting_started.html>`_ for instructions on how to create a micro-ROS environment for embedded platforms.
+
+The micro-ROS middleware, **eProsima Micro XRCE-DDS**, provides a user API that allows interfacing with the lowest level transport layer at runtime, which enables users to implement their own transports in both the micro-ROS Client and micro-ROS Agent libraries.
+
+Thanks to this, the Micro XRCE-DDS wire protocol can be transmitted over virtually any protocol, network or communication mechanism. In order to do so, two general communication modes are provided:
+
+- **Stream-oriented mode**: the communication mechanism implemented does not have the concept of packet. `HDLC framing <https://micro-xrce-dds.docs.eprosima.com/en/latest/transport.html?highlight=hdlc#custom-serial-transport>`_ will be used.
+- **Packet-oriented mode**: the communication mechanism implemented is able to send a whole packet that includes an XRCE message.
+
+These two modes can be selected by activating and deactivating the ``framing`` parameter in both the micro-ROS Client and the micro-ROS Agent functions.
+
+micro-ROS Client
+----------------
+
+An example on how to set these external transport callbacks in the micro-ROS Client API is:
+
+.. code-block:: c
+
+    #include <rmw_microros/rmw_microros.h>
+
+    ...
+
+    struct custom_args {
+        ...
+    }
+
+
+    struct custom_args args;
+
+    rmw_uros_set_custom_transport(
+        MICROROS_TRANSPORTS_FRAMING_MODE, // Framing enabled here. Using Stream-oriented mode.
+        (void *) &args,
+        my_custom_transport_open,
+        my_custom_transport_close,
+        my_custom_transport_write,
+        my_custom_transport_read
+    );
+
+
+It is important to notice that in ``rmw_uros_set_custom_transport`` a pointer to custom arguments is set. This reference will be available to every callbacks call.
+
+In general, four functions must be implemented. The behaviour of these functions is slightly different, depending on the selected mode:
+
+Open function
+^^^^^^^^^^^^^
+
+.. code-block:: c
+
+    bool my_custom_transport_open(uxrCustomTransport* transport)
+    {
+        ...
+    }
+
+
+This function should open and init the custom transport. It returns a boolean indicating if the opening was successful.
+
+``transport->args`` holds the arguments passed through ``rmw_uros_set_custom_transport``.
```

Comment:
I would move from each method section to  ​https://github.com/eProsima/vulcanexus/blob/84914038b2d015c2db46d80469ec2a7adbf72567/docs/rst/tutorials/micro/custom_transports/custom_transports.rst#L23

This way we avoid repetition.